### PR TITLE
Fix auth failure on one-way TLS

### LINKED ISSUE #1123: Federation run fails when client auth is disabled
**Describe the bug**
Using single machine and bare metal approach of Task Runner API, federation run is failing if the `disable_client_auth` flag is set to true in **plan.yaml** file (with `tls` unchanged/set as true). 

System throws `KeyError: 'x509_common_name'` from **aggregator_server.py**  (screenshot attached below)

**To Reproduce**
1. Git clone **openfl** repository, create and activate virtual environment.
2. Install all the pre-requisites.
3. **Model owner** - Create workspace `fx workspace create --template torch_cnn_mnist --prefix my_workspace` and move inside the folder.
4. **Model owner** - Modify **plan.yaml** file changing `disable_client_auth` from `false` to `true` keeping all other values intact.
5. **Model owner** - Initialise the plan `fx plan initialize`
6. **Model owner** - Certify the workspace `fx workspace certify`
7. **Aggregator** - Generate certs `fx aggregator generate-cert-request`
8. **Aggregator** - Certify the certs `fx aggregator certify -s`
9. **Collaborator1** - Create collaborator1 `fx collaborator create -n collaborator1 -d 1`
10. **Collaborator1** - Create CSR for collaborator1 `fx collaborator generate-cert-request -n collaborator1`
11. **Collaborator2** - Create collaborator2 `fx collaborator create -n collaborator2 -d 2`
12. **Collaborator2** - Create CSR for collaborator2 `fx collaborator generate-cert-request -n collaborator2`
13. **Aggregator** - Sign col1 cert request `fx collaborator certify --request-pkg col_collaborator1_to_agg_cert_request.zip -s`
14. **Aggregator** - Sign col2 cert request `fx collaborator certify --request-pkg col_collaborator2_to_agg_cert_request.zip -s`
15. **Collaborator1** - Import signed package for col1 `fx collaborator certify --import col_collaborator1_to_agg_cert_request.zip`
16. **Collaborator1** - Start collaborator1 `fx collaborator start -n collaborator1`
17. **Collaborator2** - Import signed package for col2 `fx collaborator certify --import col_collaborator2_to_agg_cert_request.zip`
18. **Collaborator2** - Start collaborator2 `fx collaborator start -n collaborator2`
19. **Aggregator** - Start aggregator `fx aggregator start`

**Expected behavior**
Federation should run smoothly.

Also, we believe that the step where aggregator certifies collaborators should be skipped as the MTLS is off (i.e. client auth is disabled). Is this the right expectation? 

It would be helpful if the exact steps to be followed in this scenario are well documented.

**Screenshots**

  **Aggregator screen**
aggregator_screen

  **Collaborator screen**
collaborator_screen

**Machine:**
 - Ubuntu 22.04   (works fine for `disable_client_auth` set to `false`)

### PR DESCRIPTION
**Pull request by MasterSkepticista**
**_Edit: Internal changes for ofl-s pending, do not merge._**

Changelog:
- [x] Fixes issue where disabling client auth with TLS-enabled, broke taskrunner experiments (#1123).
- [x] Renames `tls -> use_tls` in the plan.
- [x] Renames `disable_client_auth -> require_client_auth` in the plan. **NOTE:** This change has been made to reduce cognitive overhead of knowing how to enable mTLS in communication between the client-server. 
    * Before this PR: (`disable_client_auth=False, tls=True`) meant enabling mTLS. 
    * After this PR: (`require_client_auth=True, use_tls=True`) will enable mTLS. 
    * Default behaviour FL plans is to use mTLS.
- [x] Migrates `logger.warn -> logger.warning` to hide deprecation warnings

Tested with 4 combinations of two updated flags.

### CODE REVIEW

**Review by theakshaypant — COMMENTED** [review]
Is updating the usage of these keys in tests not in scope for this PR? [Tests Ref.](https://github.com/securefederatedai/openfl/blob/develop/tests/end_to_end/models/participants.py#L136-L137)

**Comment by MasterSkepticista** [comment]
> Is updating the usage of these keys in tests not in scope for this PR? [Tests Ref.](https://github.com/securefederatedai/openfl/blob/develop/tests/end_to_end/models/participants.py#L136-L137)

Yes. @payalcha suggested applying those changes in a separate PR (after this is merged).

File: openfl/transport/grpc/aggregator_client.py [hunk 1]
```diff
@@ -205,8 +202,8 @@ def __init__(
         Args:
             agg_addr (str): The address of the aggregator.
             agg_port (int): The port of the aggregator.
-            tls (bool): Whether to use TLS for the connection.
-            disable_client_auth (bool): Whether to disable client-side
+            use_tls (bool): Whether to use TLS for the connection.
+            require_client_auth (bool): Whether to enable client-side
                 authentication.
```
Comment (teoparvanov) [inline openfl/transport/grpc/aggregator_client.py]: ... client-side authentication _for TLS_?

File: openfl/transport/grpc/aggregator_client.py [hunk 2]
```diff
@@ -189,11 +186,11 @@ def __init__(
         self,
         agg_addr,
         agg_port,
-        disable_client_auth,
+        require_client_auth,
```
Comment (teoparvanov) [inline openfl/transport/grpc/aggregator_client.py]: same as `use_tls`, shouldn't this be `True` by default?

File: openfl/transport/grpc/aggregator_client.py [hunk 3]
```diff
@@ -288,8 +285,8 @@ def create_tls_channel(
         Args:
             uri (str): The uniform resource identifier for the secure channel.
             root_certificate (str): The Certificate Authority filename.
-            disable_client_auth (bool): True disables client-side
-                authentication (not recommended, throws warning to user).
+            require_client_auth (bool): True enables client-side
+                authentication.
```
Comment (teoparvanov) [inline openfl/transport/grpc/aggregator_client.py:289]: same here: ... client-side authentication _for TLS_?

File: openfl/transport/grpc/aggregator_server.py [hunk 4]
```diff
@@ -97,9 +95,13 @@ def validate_collaborator(self, request, context):
...
                 not authorized.
         """
-        if self.tls:
-            common_name = context.auth_context()["x509_common_name"][0].decode("utf-8")
+        if self.use_tls:
             collaborator_common_name = request.header.sender
+            if not self.require_client_auth:
+                common_name = collaborator_common_name
+            else:
+                common_name = context.auth_context()["x509_common_name"][0].decode("utf-8")
```
Comment (teoparvanov) [inline openfl/transport/grpc/aggregator_server.py]: Wouldn't this read better if reversed?
```python
  if self.require_client_auth:
      common_name = context.auth_context()["x509_common_name"][0].decode("utf-8")
  else:
      common_name = collaborator_common_name
```

File: openfl/transport/grpc/aggregator_server.py [hunk 5]
```diff
@@ -319,13 +321,13 @@ def get_server(self):
...
-            if self.disable_client_auth:
-                self.logger.warn("Client-side authentication is disabled.")
+            if not self.require_client_auth:
+                self.logger.warning("Client-side authentication is disabled.")
 
             self.server_credentials = ssl_server_credentials(
                 ((private_key_b, certificate_b),),
                 root_certificates=root_certificate_b,
-                require_client_auth=not self.disable_client_auth,
+                require_client_auth=self.require_client_auth,
```
Comment (teoparvanov) [inline openfl/transport/grpc/aggregator_server.py:330]: nice, I see where some of the inspiration for this PR came from 😉

**Review by teoparvanov — APPROVED** [review]
Great, thanks @MasterSkepticista !

PS: as discussed offline, don't forget to run the OpenFL-Security CI against this branch

File: openfl/transport/grpc/aggregator_client.py (hunk 1 above)
Comment (MasterSkepticista) [reply openfl/transport/grpc/aggregator_client.py]: Done.

File: openfl/transport/grpc/aggregator_client.py (hunk 2 above)
Comment (MasterSkepticista) [reply openfl/transport/grpc/aggregator_client.py]: Done.

File: openfl/transport/grpc/aggregator_client.py (hunk 3 above)
Comment (MasterSkepticista) [reply openfl/transport/grpc/aggregator_client.py:289]: Done.

File: openfl/transport/grpc/aggregator_server.py (hunk 4 above)
Comment (MasterSkepticista) [reply openfl/transport/grpc/aggregator_server.py]: Done.

**Review by rahulga1 — APPROVED** [review]

**Pull request merged**